src/components/WithdrawRewards/index.tsx

```diff
@@ -27,29 +29,30 @@ type WithdrawRewardsState = 'none' | 'pending' | 'success'
 function WithdrawRewards({ trigger, type, onClaim, ...rest }: WithdrawRewardsProps) {
     const { i18n } = useLingui()
     const [status, setStatus] = useState<WithdrawRewardsState>('none')
-    const { chainId } = useActiveWeb3React()
+    // const { chainId } = useActiveWeb3React() // possibly remove if not used. Method claim could return different chainId's, or array (ref > comment:117)
     const [error, setError] = useState<Error>()
     const web3 = useWeb3()
-    const [transactionHash, setTransactionHash] = useState<string>()
+    // const [transactionHash, setTransactionHash] = useState<string>() // or array (ref > comment:117)
     const dispatch = useDispatch()
     const handleClaim = useCallback(async () => {
         if (!web3) return
         try {
             setStatus('pending')
             const claimMethod = type === 'GOOD' ? claimGood : claim;
-            const transactions = await claimMethod(web3, firstTransactionHash => {
-                setTransactionHash(firstTransactionHash)
+              await claimMethod(web3, (transactionHash: string, from: string, chainId: number) => {
+                // setTransactionHash(transactionHash)
                 setStatus('success')
-            })
-            transactions.forEach(transactionDetails =>
                 dispatch(
-                    addTransaction({
-                        chainId: chainId!,
-                        hash: transactionDetails.transactionHash,
-                        from: transactionDetails.from
-                    })
+                  addTransaction({
+                    chainId: chainId!,
+                    hash: transactionHash,
+                    from: from,
+                    summary: chainId === SupportedChainId.FUSE ?
+                      i18n._(t`Claimed GOOD Rewards`) :
+                      i18n._(t`Claimed G$ Rewards`)
```

Comment:
thats not correct. you can claim GOOD on mainnet also. claimMethod/Type seems more appropriate

Reply:
yes you are right